tests/pytests/test_language.py

```diff
@@ -0,0 +1,283 @@
+    res = env.cmd('FT.SEARCH', 'idx_it', 'cherry', 'SORTBY', 'word', 'DESC',
+                  'LANGUAGE', 'english')
+    env.assertEqual(res, expected)
+    res = env.cmd('FT.SEARCH', 'idx_it', 'cherries', 'SORTBY', 'word', 'DESC',
+                  'LANGUAGE', 'english')
+    env.assertEqual(res, expected)
+
+    # Search for "cherry", without passing language argument should use the 
+    # default language of the index
+    # It should return 1 result using invalid stemming in Italian
+    res = env.cmd('FT.SEARCH', 'idx_it', 'cherry', 'SORTBY', 'word', 'ASC')
+    expected = [1, '{word}:7', ['word', 'cherry']]
+    env.assertEqual(res, expected)
+
+    # Search for "orange", without passing language argument should use the 
+    # default language of the index: Italian
+    # But in this case, the stemming in Italian, generates matching words
+    # in English
+    res = env.cmd('FT.SEARCH', 'idx_it', 'orange', 'SORTBY', 'word', 'ASC')
+    expected = [2, '{word}:5', ['word', 'orange'], '{word}:6', ['word', 'oranges']]
+    env.assertEqual(res, expected)
+
+    # Search for "ciliegia", the document was created without __lang value
+    # but during the data ingestion the terms were created using the index 
+    # language: Italian
+    expected = [2, '{word}:3', ['word', 'ciliegia'], '{word}:4', ['word', 'ciliegie']]
+    res = env.cmd('FT.SEARCH', 'idx_it', 'ciliegia', 'SORTBY', 'word', 'ASC')
+    env.assertEqual(res, expected)
+    res = env.cmd('FT.SEARCH', 'idx_it', 'ciliegie', 'SORTBY', 'word', 'ASC')
+    env.assertEqual(res, expected)
+
+    # Create index without language per index
+    env.cmd('FT.CREATE', 'idx_en', 'ON', 'HASH', 'PREFIX', '1', '{word}:',
+            'LANGUAGE_FIELD', '__lang', 'SCHEMA', 'word', 'TEXT', )
+    # Wait for index to be created
+    waitForIndex(env, 'idx_en')
+
+    # Search for "cherry/cherries", without passing language argument, should
+    # use the default language: English
+    # It should return 2 results using stemming in English
+    expected = [2, '{word}:7', ['word', 'cherry'], '{word}:8', ['word', 'cherries']]
+    res = env.cmd('FT.SEARCH', 'idx_en', 'cherry', 'SORTBY', 'word', 'DESC')
+    env.assertEqual(res, expected)
+    res = env.cmd('FT.SEARCH', 'idx_en', 'cherries', 'SORTBY', 'word', 'DESC')
+    env.assertEqual(res, expected)
+
+    # Search for "arance/arancia", "ciliegia/ciliegie", without passing language
+    # argument, should use language by default: English
+    # To validate this, we check that we have the same results in both cases:
+    # 1. passing the language argument = English
+    # 2. searching without language argument
+    for words in ['arancia', 'arance', 'ciliegia', 'ciliegie']:
+        res1 = env.cmd('FT.SEARCH', 'idx_en', 'arance', 'SORTBY', 'word', 'DESC')
+        res2 = env.cmd('FT.SEARCH', 'idx_en', 'arance', 'SORTBY', 'word', 'DESC',
+                        'LANGUAGE', 'english')
+        env.assertEqual(res1, res2)
+
+    
```

Comment:
Consider testing using the lang field such as 
`@lang:(italian)  ciliegia LANGUAGE ITALIAN`
Or
`@lang:(italian)  ciliegia` with index language `Italian` ?

Reply:
New tests included in [2b3dbc2](https://github.com/RediSearch/RediSearch/pull/4455/commits/2b3dbc24fbde982dde73e1de38282cb88e497f21)

Reply:
Tests included in [2b3dbc2](https://github.com/RediSearch/RediSearch/pull/4455/commits/2b3dbc24fbde982dde73e1de38282cb88e497f21)